lib/ash_json_api/resource/transformers/require_primary_key.ex

```diff
@@ -0,0 +1,37 @@
+defmodule AshJsonApi.Resource.Transformers.RequirePrimaryKey do
+  @moduledoc "Ensures that the resource either has a primary key 
+    or includes primary_key section from json_api if it has a composite key
+  "
+
+  use Ash.Dsl.Transformer
+  alias Ash.Dsl.Transformer
+
+  def transform(_resource, dsl) do
+    case Transformer.get_option(dsl, [:json_api, :primary_key], :keys) do
+      nil ->
+        dsl
+        |> Transformer.get_entities([:attributes])
+        |> Enum.filter(& &1.primary_key?)
+        |> Enum.map(& &1.name)
+        |> case do
+          [_only_one_primary_key] -> {:ok, dsl}
+          _ -> raise "AshJsonApi requires primary key when a resource has a composite key"
```

Comment:
Lets use the newer DslError

Here is an example:
```elixir
      raise Ash.Error.Dsl.DslError,
        module: __MODULE__,
        path: [:relationships, relationship.name],
        message:
          "Relationship `#{relationship.name}` expects source field `#{relationship.source_field}` to be defined"
```
I know this is based off of my code from the other transformer, so its my fault :D

Reply:
Added Ash.Error.Dsl.DslError